torch/utils/data/datapipes/dataframe/dataframes.py

```diff
@@ -0,0 +1,292 @@
+
+from torch.utils.data.datapipes.dataframe.structures import DataChunkDF
+
+# TODO(VitalyFedyunin): Add error when two different traces get combined
+class DataFrameTracedOps(DFIterDataPipe):
+    def __init__(self, source_datapipe, output_var):
+        self.source_datapipe = source_datapipe
+        self.output_var = output_var
+
+    def __iter__(self):
+        for item in self.source_datapipe:
+            yield self.output_var.calculate_me(item)
+
+
+#  TODO(VitalyFedyunin): Extract this list from the DFIterDataPipe registred functions
+DATAPIPES_OPS = ['_dataframes_as_tuples', 'groupby', '_dataframes_filter', 'map', 'to_datapipe',
+                 'shuffle', 'concat', 'batch', '_dataframes_per_row', '_dataframes_concat', '_dataframes_shuffle']
+
+
+class Capture(object):
+    # All operations are shared across entire InitialCapture, need to figure out what if we join two captures
+    ctx: Dict[str, List[Any]]
+
+    def __init__(self):
+        self.ctx = {'operations': [], 'variables': []}
+
+    def __str__(self):
+        return self.ops_str()
+
+    def ops_str(self):
+        res = ""
+        for op in self.ctx['operations']:
+            if len(res) > 0:
+                res += "\n"
+            res += str(op)
+        return res
+
+    def __getattr__(self, attrname):
+        if attrname == 'kwarg':
+            raise Exception('no kwargs!')
+        if attrname in DATAPIPES_OPS:
+            return (self.as_datapipe()).__getattr__(attrname)
+        return CaptureGetAttr(self, attrname, ctx=self.ctx)
+
+    def __getitem__(self, key):
+        return CaptureGetItem(self, key, ctx=self.ctx)
+
+    def __setitem__(self, key, value):
+        self.ctx['operations'].append(
+            CaptureSetItem(self, key, value, ctx=self.ctx))
+
+    def __add__(self, add_val):
+        res = CaptureAdd(self, add_val, ctx=self.ctx)
+        var = CaptureVariable(res, ctx=self.ctx)
+        self.ctx['operations'].append(
+            CaptureVariableAssign(variable=var, value=res, ctx=self.ctx))
+        return var
+
+    def __sub__(self, add_val):
+        res = CaptureSub(self, add_val, ctx=self.ctx)
+        var = CaptureVariable(res, ctx=self.ctx)
+        self.ctx['operations'].append(
+            CaptureVariableAssign(variable=var, value=res, ctx=self.ctx))
+        return var
+
+    def __mul__(self, add_val):
+        res = CaptureMul(self, add_val, ctx=self.ctx)
+        var = CaptureVariable(res, ctx=self.ctx)
+        t = CaptureVariableAssign(variable=var, value=res, ctx=self.ctx)
+        self.ctx['operations'].append(t)
+        return var
+
+    def as_datapipe(self):
+        return DataFrameTracedOps(
+            self.ctx['variables'][0].source_datapipe, self)
+
+    def raw_iterator(self):
+        return self.as_datapipe().__iter__()
+
+    def __iter__(self):
+        return iter(self._dataframes_as_tuples())
+
+    def batch(self, batch_size=10):
+        dp = self._dataframes_per_row()._dataframes_concat(batch_size)
```

Comment:
Curious on this: After `_dataframes_per_row` , the `DataPipe` will return one row a t a time (so it flatten the DataFrame`). 

But`_dataframes_concat`  will concat multiple DataFrames together?